### PR TITLE
Jt/visually segmented case search groups

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

This change affects `queryResponse`. It adds `groupKey` to `display` elements and `groupHeaders` dictionary to `queryResponse`. The `groupKey` of each display element corresponds to a key in `groupHeaders`.
## Technical Summary

[Tech Spec](https://docs.google.com/document/d/1syEA-HC5g1FOjZ6m8DlfBspzKn9hKvq0zbMNbIWzQVE/edit#heading=h.d7s2sycdfk92)
[Jira Ticket](https://dimagi-dev.atlassian.net/browse/USH-3826)
[Research Doc](https://docs.google.com/document/d/1ZkN0_ta8jEjQ5IYHLIduBN6ZFcVEjfSIQrN3AQgBVgw/edit#heading=h.x97repjoa0f2)

Commcare-core PR: https://github.com/dimagi/commcare-core/pull/1389
HQ PR: https://github.com/dimagi/commcare-hq/pull/33828
### Safety story

locally tested.
This does not alter existing response values but only adds new key:values.  

### Automated test coverage

Will add tests

### QA Plan

Will [QA](https://dimagi-dev.atlassian.net/browse/QA-5881)

### Special deploy instructions
Needs to be merged with/after https://github.com/dimagi/commcare-core/pull/1389

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations.

### Review

- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

**Review by nospame — APPROVED** [review]
Reviewed with Robert, looks good!

**Review by shubham1g5 — CHANGES_REQUESTED** [review]
Couple Minor comments. 

In Addition, we should update the `libs/commcare` to point to the head of FP branch in this PR before you merge it (after merging the corresponding core PR) so that we don't overwrite any changes that got to `libs/commcare` after you update the submodule. The practice I usually follow here is to not update the submodule at all in the formplayer PR until I am about to merge it. When the Prs are ready to be merged I do the following sequence of events - 

1. Merge `commcare-core` PR into `formplayer` branch
2. Update submodule in `Formplayer` PR to point to head of `formplayer` branch in `commcare-core`
3. Wait for tests to pass in formplayer PR and merge if the build passes.

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java]: [11d6ed6](https://github.com/dimagi/formplayer/pull/1511/commits/11d6ed6732f1c192a4c2d7ac19a66aba435f2f64) and [15a0d9e](https://github.com/dimagi/commcare-core/pull/1389/commits/15a0d9ecbc5783cc28d53531e81312208866f4f9)

Can you explain what the responsibility of `queryScreen` is? Why would the preference be to put the logic in there instead of extracting it to another function in `queryResponseBean`? I would imagine `queryResponesBean` is responsible for generating the data in the response.

File: src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java]: I couldn't tell which convention is "correct". We seem to use both, `getItemsetChoices`. I'm guessing camelcase makes more sense since that's the convention for Java and javascript (which uses the data).

But you're right, it can be removed [af803fd](https://github.com/dimagi/formplayer/pull/1511/commits/af803fdfd3eb7c29bdeb8f3640b386f7d343ba23)

File: src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/DisplayElement.java]: discussed in other comment thread https://github.com/dimagi/formplayer/pull/1511#discussion_r1424757506

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java]: `queryResponseBean` is supposed to me more of a response model wrapper class over `screen` classes with `screen` classes being the classes that contain any evaluation logic. We don't follow it strictly but that's the idea atleast in theory.

**Review by shubham1g5 — APPROVED** [review]
Approved (but core submodule need to be updated before merge)

**Pull request merged**
